### DISCUSSION TITLE
Get aux heater status and power with EMS-ESP

### DISCUSSION THREAD

**Discussion by kriztan** [emsesp/EMS-ESP32#1160] [Ideas]
I'm using the EMS-ESP 3.5.1 with my Bosch Compress 7400i heat pump and it would be nice to get the status and power of the internal electrical (aux) heater. Currently the "auxilliary heater status" is alway 0 and I can't find an entry with shows the current power.

**Comment by MichaelDvP** [comment]
We have this question before, but it needs someone with this heatpump to log the telegrams and find the value. 
You have to log all while the heater switches to see the change. I expect it somewhere in the 0x048D telegram, but not sure. From #535 we know that husdata shows the heater as 0/33/66/100%,. Could be a percentage value in telegram, or switching 3 phases, represented by 3 bits in a data position. If you post the log and can describe at which time the heater switches to which level, we can help to find the position.

**Reply by kriztan** [reply]
Thank for your reply. But I don't know how to force a special value/setting in the heatpump. I'll have to read a bit more.

**Reply by MichaelDvP** [reply]
I don't have a HP, but as far as i know the aux heater is a direct electrical heater, triggert
1. if heat demand is higher than power of heating pump
2. if temperature setting is higher than max temp of pump.
3. emerency operation if pump is not running safe.

With actual outside temperatures the pump power is enough, so the best way to trigger is imo 2.) maybe something like dhw disinfection with 70°C. Or disconnect a pressure sensor sensor to set the pump in emergency operation.

**Reply by MichScha** [reply]
@kriztan it is like @MichaelDvP wrote - just set the heating window for the heatpump to a timeframe that doesn´t match the desinfect timeslot and activate desinfect with ems-esp or directly on your heatpump. The aux-heater will start as soon as the set timeframe of desinfect reaches the current clock. To raise the aux-power just increase also the limits - for my wlw196 I can choose between different values.

**Reply by seh-len** [reply]
I have attached a log of the various stages of my WLW196 electric auxiliary heater during single charge.
The auxiliary heater supports the steps [0, 2, 4, 6, 9] kW. When changing from 2 kW to 4 kW, the relay for the auxiliary heater temporarily switches off completely and then switches on 4 kW.
[switching el heater.log](https://github.com/emsesp/EMS-ESP32/files/14393500/switching.el.heater.log)

**Comment by binarious** [comment]
@kriztan Sorry to hijack your thread, but I haven't found another way contacting you.

Since I'd like to control the same heat pump: Would you mind to share your setup (where and how to connect the gateway) and what kind of data you receive/can change?

Thanks in advance!

**Comment by maecki-maecki** [comment]
I own a new CS6800i and a bbqkees Gateway.
boiler.auxheaterstatus is always off in ems-esp, even when the display at the unit shows it is activated.

Is this something that is already worked on with the log file of seh-len or do you need an additional dump? If the latter applies plz point me to how I can get the registers dumped while I switch the aux heater on… Interpreting ems traffic is beyond my knowledge, though…

**Reply by maecki-maecki** [reply]
Don’t mind - I just found the ‚all’ switch in logging mode  ;-)

[log 6.txt](https://github.com/emsesp/EMS-ESP32/files/15077128/log.6.txt)

around
14:29:59  I activated the extra dhw in the bosch app, hp started soon after
14:41   aux heater jumped in
14:43  I deactivated the extra dhw in the app, aux heater switched off soon after, hp also

Hope this helps to interpret aux elec heater datagrams… @MichaelDvP ?

btw my aux heater is limited to 3kW (possible settings are 0,3,6 or 9kW)
ems-esp is 3.6.5

**Reply by maecki-maecki** [reply]
My guess:

HpPower(0x048D), data: 11 3B 00 21 03 03 1E (offset 3)
Fourth Byte (plus 3 offset) set to 33% = 0x21

HpPower(0x048D), data: 10 64 00 00 03 00 00 00 1B 00 00 00 00 38 4B 4B 3C 00 2C 14 (offset 3)
Fourth Byte (plus 3 offset) set to 0% = 0x00

This would align with @seh-len 's log above and his observation that between 22% (2kW) and 44%(4kW) it falls back to 0 (22% is 0x16, 44% is 0x2c)

004+07:05:40.361 N 450: [emsesp] boiler(0x08) -W-> Me(0x0B), HpPower(0x048D), data: 81 00 10 09 23 00 16 00 03 14 00 00 04 00 00 00 00 00 00 00 00 10 00
004+07:05:55.961 N 452: [emsesp] boiler(0x08) -B-> All(0x00), HpPower(0x048D), data: 81 00 10 09 23 00 00 00 03 00 00 00 04 00 00 00 00 00 00 00 00 10 00
004+07:05:58.761 N 453: [emsesp] boiler(0x08) -B-> All(0x00), HpPower(0x048D), data: 81 00 10 09 23 00 2C 00 03 28 00 00 04 00 00 00 00 00 00 00 00 10 00
0


So 0x048D offset 7, one byte as percentage might be the aux elec heater status

@MichaelDvP : worth a shot?

**Reply by MichaelDvP** [reply]
Looks good, have you tried to create a custom entity and check the states. 
![grafik](https://github.com/emsesp/EMS-ESP32/assets/59284019/807d989c-020e-4e55-8d25-b64b5a5c2f7d)
It's offset 6, because we start from 0 and offset 3 means the firste byte is offset 3.
I'll add in next dev or dev2.

**Reply by maecki-maecki** [reply]
Just tested the custom entity and it works like a charm…
Maybe @seh-len might try this too?
Thanks!

**Comment by mtbsteve** [comment]
@MichaelDvP has this been added yet or do I still need to creatre a custom entity? I am in version 3.6.5 and the aux_heater_status entity does not get triggered when the aux heater is switched on. I have a Buderus 196i heatpump.

**Reply by proddy** [reply]
Yes, Michael added this to 3.7.0 5 months ago. See https://github.com/emsesp/EMS-ESP32/pull/1711

**Reply by mtbsteve** [reply]
Thanks! I look forward to the 3.7 release.
